### ISSUE TITLE
`NullPointerException` in `DTDValidator.validateElementEnd()` for element undefined in DTD

### ISSUE THREAD

**Issue by ChrisTrenkamp** [FasterXML/woodstox#104]
The DTDValidator class is throwing a NullPointerException when processing invalid XML.  To reproduce, use this catalog: https://github.com/oasis-tcs/dita/tree/v1.3/doctypes

test.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">
<map>
  <topicref>
    <ditavalref>
      <val><!-- This is invalid -->
        <prop att="product" val="win" action="flag" color="black"/>
      </val>
    </ditavalref>
  </topicref>
</map>
```

WstxValidatorTest.java
```java
import com.ctc.wstx.stax.WstxInputFactory;
import java.io.File;
import java.nio.file.Paths;
import javax.xml.catalog.CatalogFeatures;
import javax.xml.catalog.CatalogManager;
import javax.xml.stream.XMLInputFactory;
import javax.xml.stream.XMLStreamException;

public class WstxValidatorTest {
  public static void main(String[] args) throws XMLStreamException {
    var catalog =
        CatalogManager.catalogResolver(
            CatalogFeatures.defaults(), Paths.get("path/to/catalog.xml").toUri());
    var factory = new WstxInputFactory();
    factory.setProperty(XMLInputFactory.IS_VALIDATING, true);
    factory.setProperty(XMLInputFactory.RESOLVER, catalog);

    var reader = factory.createXMLStreamReader(new File("path/to/test.xml"));
    reader.setValidationProblemHandler(p -> System.out.println(p.getMessage()));

    while (reader.hasNext()) {
      reader.next();
    }
  }
}
```

```
Undefined element <val> encountered
Undefined element <prop> encountered
Exception in thread "main" java.lang.NullPointerException
	at com.ctc.wstx.dtd.DTDValidator.validateElementEnd(DTDValidator.java:362)
	at com.ctc.wstx.sr.InputElementStack.validateEndElement(InputElementStack.java:560)
	at com.ctc.wstx.sr.BasicStreamReader.nextFromTree(BasicStreamReader.java:2730)
	at com.ctc.wstx.sr.BasicStreamReader.next(BasicStreamReader.java:1122)
	at WstxValidatorTest.main(WstxValidatorTest.java:22)
```

**Comment by cowtowncoder** [comment]
Thank you for reporting the issue; sounds like a bug.

**Comment by cowtowncoder** [comment]
Hmmh. One problem: `javax.xml.catalog` is JDK 9 only... need to figure out how to test that; Woodstox itself only requires JDK 6 (although I am ok with JDK 8 for build, just not versions beyond that).
Will try to see if this might be part of some additional library before JDK 9.

**Comment by cowtowncoder** [comment]
Ok can probably just implement `XMLResolver` directly, use code otherwise as-is.

**Comment by cowtowncoder** [comment]
Except... this requires catalog resolution. Will try to see if I can temporarily use different JDK and see if I can at least temporarily reproduce the issue.

**Comment by cowtowncoder** [comment]
Ok no. I just can't make the test work as described. Catalog resolves throws

    javax.xml.catalog.CatalogException: JAXP09040001: No match found for publicId '-//OASIS//DTD DITA Map//EN' and systemId 'map.dtd'.

regardless of how I try to initialize it.

So unfortunately I can not replicate the problem at this point.

**Comment by ChrisTrenkamp** [comment]
I get that error when the path to the catalog is incorrect in the test case.  Here's a new catalog to try.

[doctypes.zip](https://github.com/FasterXML/woodstox/files/4515906/doctypes.zip)

**Comment by cowtowncoder** [comment]
Thank you! Now I can reproduce this locally, and it seems to be due to undefined element, for which `validateElementStart()` does not add matching `StructValidator` (since it is undefined).
Should now be able to create simpler test to reproduce, then see what is the best way to resolve.

**Comment by cowtowncoder** [comment]
Fixed for next release (either 6.1.2 or 6.2.0, depending a bit on if #103 fix is included and needs minor version bump); also backported in 5.3 branch just in case there's need for release there.
